go/libraries/doltcore/sqle/tables.go

```diff
@@ -1803,3 +1794,69 @@ func (t *AlterableDoltTable) DropPrimaryKey(ctx *sql.Context) error {
 
 	return t.updateFromRoot(ctx, newRoot)
 }
+
+// canDropPk checks whether there are conflicting foreign key relationships referencing this
+// table's primary key tags
+func (t *AlterableDoltTable) verifyCanDropPk(ctx *sql.Context, root *doltdb.RootValue) (*doltdb.ForeignKeyCollection, error) {
```

Comment:
Don't mix business logic and basic data access functionality like this. The fact that this method is called "verifyX" but returns a FKC (and in fact replaces a method called GetForeignKeyCollection in the usage above) is a huge red flag. This is a bad factoring, find a way to cleanly separate these two concerns. May be as simple as only returning a bool or err here instead of a FKC, but I haven't read very closely

Reply:
yeah makes sense. the data structures i have to build for this seem expensive, so not sure we want to do it twice. `updateFkForNewSchema`, `updateFkForPkDrop`, `fkBackupsForPkDrop`, or something. happy to split search and modify into two steps if need be, seems equivalent on my end